do not delete MicroServiceChainLinks

jhsimpson opened the pull request:
refs #8414
refs #8039
refs #6488
refs #7239
refs #7922
refs #8019

updated 4 mysql_dev*sql scripts, removing lines that
delete from MicroServiceChainLinks.  These delete statements
can fail when run on an Archivematica server that already has
Jobs in its database that reference the MCSL's.

The fail prevents upgrading from completing properly.

jhsimpson committed:
do not delete MicroServiceChainLinks

refs #8414
refs #8039
refs #6488
refs #7239
refs #7922
refs #8019

updated 4 mysql_dev*sql scripts, removing lines that
delete from MicroServiceChainLinks.  These delete statements
can fail when run on an Archivematica server that already has
Jobs in its database that reference the MCSL's.

The fail prevents upgrading from completing properly.

mistydemeo commented:
:+1:

Hwesta commented:
Can you comment the lines out instead of deleting them?

I wish we could have a try-catch around the deletes, but I know SQL makes that hard.

jhsimpson commented:
What about collecting the deleted deletions and storing them in a separate sql file?  We can try to delete them, in a separate step, but if that fails, then at least it won't bork the upgrade.

Hwesta commented:
I really like that idea.  I don't want the MSCLs cluttering up the DB if they don't have to, and the separate file for deletions would handle that on upgrade.

jhsimpson committed:
Keep list of old Chain Links to delete

refs #8414

deletion of deprecated MicroServiceChainLinks is moved to the last step of
database upgrades, to ensure it doesn't prevent any useful sql from executing.

mistydemeo commented:
How about making the deletions a separate script entirely? `mysql_dev_cleanup.sh` or some such.

Hwesta commented on src/MCPServer/share/dev_mysql_delete_links.sql:
```diff
@@ -0,0 +1,92 @@
+DELETE FROM MicroServiceChainLinks WHERE pk IN (@del);
+
+SET @d1 = '2307b24a-a019-4b5b-a520-a6fff270a852' COLLATE utf8_unicode_ci;
+SET @d2 = 'c4e109d6-38ee-4c92-b83d-bc4d360f6f2e' COLLATE utf8_unicode_ci;
```
These two should be

```
SET @d1 = '4df4cc06-3b03-4c6f-b5c4-bec12a97dc90' COLLATE utf8_unicode_ci;
SET @d2 = '5e4f7467-8637-49b2-a584-bae83dabf762' COLLATE utf8_unicode_ci;
```

instead of duplicating the ones below.

Hwesta commented on src/MCPServer/share/dev_mysql_delete_links.sql:
```diff
@@ -0,0 +1,92 @@
+DELETE FROM MicroServiceChainLinks WHERE pk IN (@d1, @d2, @d3, @d4, @d5);
+
+SET @removeIndexedAIPFiles='bfade79c-ab7b-11e2-bace-08002742f837' COLLATE utf8_unicode_ci;
+DELETE FROM MicroServiceChainLinks WHERE pk=@removeIndexedAIPFiles;
```
Also needs

```
SET @indexAIPfiles='3ba518ab-fc47-4cba-9b5c-79629adac10b' COLLATE utf8_unicode_ci;
DELETE FROM MicroServiceChainLinks WHERE pk=@indexAIPfiles;
```

mistydemeo commented:
A reason I'd be interested in doing that is because I want to start migrating our bash scripts to use [`set -e`](http://redsymbol.net/articles/unofficial-bash-strict-mode/), which will cause the entire script to fail if an expression returns non-zero. If we do that, it might be better to keep something expected to fail in another file where we can treat it specially.

jhsimpson commented:
I added the missing delete statements, and fixed the name of the file holding the delete statements, so it matches how it is called in mysql_dev.sh, and squashed those changes together, so this PR now has only 2 commits.

Hwesta commented:
:+1:
